library/ecp.c

```diff
@@ -2059,6 +2064,60 @@ int mbedtls_ecp_check_pub_priv( const mbedtls_ecp_keypair *pub, const mbedtls_ec
     return( ret );
 }
 
+#if defined(MBEDTLS_ASN1_WRITE_C) && defined(MBEDTLS_OID_C)
+int mbedtls_ecp_ansi_write_group( const mbedtls_ecp_group *grp,
+                                  unsigned char *p,
+                                  size_t size, size_t *olen )
+{
+    const char *oid;
+    unsigned char *q;
+    size_t oid_length;
+    int ret;
+
+    ret = mbedtls_oid_get_oid_by_ec_grp( grp->id, &oid, &oid_length );
+    if( ret != 0 )
+        return( ret );
+    // Output is a TLV with len(T)=1, len(L)=1, V=OID
+    if( size < 2 + oid_length )
+        return( MBEDTLS_ERR_ECP_BUFFER_TOO_SMALL );
+    q = p + 2 + oid_length;
+    *olen = mbedtls_asn1_write_oid( &q, p, oid, oid_length );
```

Comment:
We don't obey the return code here. Suggestion: Initialize `*olen = 0` and use `MBEDTLS_ASN1_CHK_ADD( *olen, mbedtls_asn1_write_oid( ... ) );`.

Reply:
Done.

Reply:
@gilles-peskine-arm @mpg @yanesca Ping, this is a functional change since your last approval.

Reply:
@hanno-arm Is there anything else you would like to have reworked? :)